lib/metadata/metadataUtils.js

```diff
@@ -62,11 +62,16 @@ function getNullVersionFromMaster(bucketName, objectKey, log, cb) {
  * @param {string} [versionId] - version of object to retrieve
  * @param {RequestLogger} log - request logger
  * @param {function} cb - callback
+ * @param {object} cachedDocument - cached version of the document used for
+ *                                  abstraction purposes
  * @return {undefined} - and call callback with err, bucket md and object md
  */
-function metadataGetObject(bucketName, objectKey, versionId, log, cb) {
+function metadataGetObject(bucketName, objectKey, versionId, log, cb, cachedDocument = null) {
```

Comment:
shouldn't `cb` be the last param?

Reply:
also IMO we should pass the whole cache to this function, and let it locate the object : if the caller already handles finding the object in the cache, then he may just as well skip the function altogether...

Reply:
> shouldn't cb be the last param?

We can make it a non optional parameter yes, and change all calls to this function...